test/cli.zig

```diff
@@ -96,18 +100,65 @@ fn exec(cwd: []const u8, expect_0: bool, argv: []const []const u8) !ChildProcess
 }
 
 fn testZigInitLib(zig_exe: []const u8, dir_path: []const u8) !void {
-    _ = try exec(dir_path, true, &[_][]const u8{ zig_exe, "init-lib" });
-    const test_result = try exec(dir_path, true, &[_][]const u8{ zig_exe, "build", "test" });
+    _ = try exec(dir_path, 0, &[_][]const u8{ zig_exe, "init-lib" });
+    const test_result = try exec(dir_path, 0, &[_][]const u8{ zig_exe, "build", "test" });
     try testing.expectStringEndsWith(test_result.stderr, "All 1 tests passed.\n");
 }
 
 fn testZigInitExe(zig_exe: []const u8, dir_path: []const u8) !void {
-    _ = try exec(dir_path, true, &[_][]const u8{ zig_exe, "init-exe" });
-    const run_result = try exec(dir_path, true, &[_][]const u8{ zig_exe, "build", "run" });
+    _ = try exec(dir_path, 0, &[_][]const u8{ zig_exe, "init-exe" });
+    const run_result = try exec(dir_path, 0, &[_][]const u8{ zig_exe, "build", "run" });
     try testing.expectEqualStrings("All your codebase are belong to us.\n", run_result.stderr);
     try testing.expectEqualStrings("Run `zig build test` to run the tests.\n", run_result.stdout);
 }
 
+fn testZigCC(zig_exe: []const u8, dir_path: []const u8) !void {
+    // Test writer was not able to figure out how to start a child process and
+    // give an fd to it's stdin. fork/exec works, but we are limiting ourselves
+    // to POSIX.
+    if (builtin.os.tag == .windows)
+        // no error.SkipZigTest, because this is not a "test {...}" block.
+        return;
+
+    return testZigCCPosix(zig_exe, dir_path);
+}
+
+fn testZigCCPosix(zig_exe: []const u8, dir_path: []const u8) !void {
+    var tmp = testing.tmpDir(.{});
+    defer tmp.cleanup();
+    try tmp.dir.writeFile("truth.c", "int main() { return 42; }");
```

Comment:
Why is using child_proess.zig no option here?

Reply:
That's written in https://github.com/ziglang/zig/pull/14462/files#diff-762da4372d6d5407ded5b08627e8ee3400f9687590ab42ab43eb1246f8bf5646R116-R118 : in case that was not clear,  I wasn't able to spawn a `ChildProcess` whose stdin would be custom fd.

Does that explain it?

Reply:
I'm not sure, if lib/std/child_process should encode specifically all the things one can do with file descriptors to standard streams or where specifically we set the limit, because handling Windows without child_process.zig is extremely verbose.

My PR https://github.com/ziglang/zig/pull/14152 simplifies serializing and deserialization the handle=fd, but is still more verbose (requires child process to have parsing stdin/environment). Would that be an option?
Otherwise it would be nice, if you can leave a review on what I can improve.